saleor/checkout/fetch.py

```diff
@@ -137,6 +154,17 @@ def fetch_checkout_lines(checkout: "Checkout") -> Iterable[CheckoutLineInfo]:
                 collections=collections,
             )
         )
+
+    if validate_variants and unavailable_variant_pks:
+        not_available_variants_ids = {
+            graphene.Node.to_global_id("ProductVariant", pk)
+            for pk in unavailable_variant_pks
+        }
+        raise ValidationError(
+            "Unavailable variants.",
+            params={"variants": not_available_variants_ids},
+        )
```

Comment:
I would remove the flag argument and instead return two lists from this function (one with available and the other with unavailable lines). The mutations should handle raising the validation error.